frontend/src/metabase/dashboard/dashboard.js

```diff
@@ -1173,6 +1160,30 @@ const loadingDashCards = handleActions(
   { dashcardIds: [], loadingIds: [], startTime: null },
 );
 
+const loadMetadataForDashboard = dashCards => (dispatch, getState) => {
+  const metadata = getMetadata(getState());
+  _.chain(dashCards)
+    .map(dc => [dc.card].concat(dc.series))
+    .flatten()
+    .map(card => new Question(card, metadata).query().dependentMetadata())
```

Comment:
I think everything from here down (except the `new Question().query()` part) should go in frontend/src/metabase/redux/metadata.js in a function called `loadMetadataForQueries` since they're similar. We could even replace `loadMetadataForQuery` with `const loadMetadataForQuery = query => loadMetadataForQueries([query])`

Reply:
e.x. something like


```js
export const loadMetadataForQuery = query => loadMetadataForQueries([query]);

export const loadMetadataForQueries = query => dispatch => {
  const dependencies = queries
    .map(query => query.dependentMetadata())
    .flatten()
    // these next three lines dedupe deps by (type, id)
    .groupBy(dm => dm.type + dm.id)
    .values()
    .map(([dependency]) => dependency)
  return Promise.all(
    dependencies.map(({ type, id, ...options }) => {
      if (type === "table") {
        if (options.foreignTables) {
          return dispatch(Tables.actions.fetchMetadataAndForeignTables({ id }));
        } else {
          return dispatch(Tables.actions.fetchMetadata({ id }));
        }
      } else {
        console.warn(`loadQueryMetadata: type ${type} not implemented`);
      }
    }),
  ).catch(e => console.error("Failed loading metadata for query", e));
  )
}

Reply:
Then 

```js
const queries = getMetadata(getState());
  _.chain(dashCards)
    .map(dc => [dc.card].concat(dc.series))
    .flatten()
    .map(card => new Question(card, metadata).query())
dispatch(loadMetadataForQueries(queries));
```

Reply:
👍 I like it. I pushed those changes.